### PR TITLE
Add conversion of existing library to GNPS batch library submission

### PR DESCRIPTION
**Pull request by robinschmid**
- Import library
- export batch submission formats: mgf and tsv

TODO 
- add direct link to validator (https://gnps-quickstart.ucsd.edu/validatebatch)
- instruct on submission

### CODE REVIEW

File: src/main/java/io/github/mzmine/datamodel/features/types/annotations/PeptideSequenceType.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+public class PeptideSequenceType extends StringType implements EditableColumnType,
+    StringParser<String>, AnnotationType {
+
+  private StringConverter<String> converter = new DefaultStringConverter();
+
+  @NotNull
+  @Override
+  public final String getUniqueID() {
+    // Never change the ID for compatibility during saving/loading of type
+    return "peptide";
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/datamodel/features/types/annotations/PeptideSequenceType.java:45]: peptide_sequence

File: src/main/java/io/github/mzmine/modules/io/export_library_gnps_batch/GNPSLibraryBatchExportTask.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        setErrorMessage("Cannot create directory for file " + mgfFile);
+        return;
+      }
+    } catch (Exception e) {
+      setStatus(TaskStatus.ERROR);
+      setErrorMessage("Cannot create directory for file " + mgfFile);
+      return;
+    }
+
+    boolean writeTsvHeader = !tsvFile.exists();
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/io/export_library_gnps_batch/GNPSLibraryBatchExportTask.java:112]: The parameter says it existing files will be overwritten, are you sure you dont want to always write the header? Or is the description wrong.?

File: src/main/java/io/github/mzmine/modules/io/spectraldbsubmit/formats/MGFEntryGenerator.java [hunk 3]
```diff
@@ -156,7 +166,12 @@ public static String createMGFEntry(SpectralLibraryEntry entry) {
       if (id == null || id.isBlank()) {
         continue;
       }
-      entry.getField(field).ifPresent(value -> appendValue(s, field, value));
+      if (scanNumber != null && (field == DBEntryField.SCAN_NUMBER
+                                 || field == DBEntryField.FEATURE_ID)) {
+        appendValue(s, field, scanNumber);
+      } else {
+        entry.getField(field).ifPresent(value -> appendValue(s, field, value));
+      }
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/io/spectraldbsubmit/formats/MGFEntryGenerator.java:174]: what's the difference between these two cases? do oyu just want the scan number as feature id and as scan number?

Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/io/spectraldbsubmit/formats/MGFEntryGenerator.java:174]: added some comments. GNPS needs both to be the scan number
```java
      // if scanNumber override is set - replace scan number and featureID (used by GNPS)
      if (scanNumber != null && (field == DBEntryField.SCAN_NUMBER
                                 || field == DBEntryField.FEATURE_ID)) {
        appendValue(s, field, scanNumber);
      } else {
        // just use the value
        entry.getField(field).ifPresent(value -> appendValue(s, field, value));
      }
```

**Comment by robinschmid** [comment]
revisions done

**Pull request merged**